quartz_solar_forecast/forecasts/v2.py

```diff
@@ -0,0 +1,234 @@
+import datetime
+import pandas as pd
+import zipfile
+import gdown
+import os.path
+
+from quartz_solar_forecast.weather import WeatherService
+
+from xgboost.sklearn import XGBRegressor
+
+
+class TryolabsSolarPowerPredictor:
+    """
+    A class to predict solar power output based on weather data, location, panel orientation,
+        and other factors.
+
+    Attributes
+    ----------
+    model_path : str
+        Path to the trained model joblib file.
+
+    Methods
+    -------
+    load_model -> None:
+
+        Downloads the model from Google Drive and decropresses it if necessary
+    
+    predict_power_output(latitude: float, longitude: float, start_date: str, kwp: float,
+        orientation: float, tilt: float) -> pd.DataFrame:
+
+        Predicts solar power output for the given parameters.
+    """
+    DATE_COLUMN = "date"
+
+    def _download_model(self, filename: str, file_id: str) -> None:
+        """
+        Download model from google drive.
+
+        Parameters
+        ----------
+        filename : str
+            The name of the model to be saved
+        file_id: 
+            Google id of the model file
+        """
+        gdown.download(f'https://drive.google.com/uc?id={file_id}', filename, quiet=False)
+
+    def _decompress_zipfile(self, filename: str) -> None:
+        """
+        Extract all files contained in a .zip file to the current directory.
+        filename must contain .zip extension
+
+        Parameters
+        ----------
+        filename : str
+            The name of the .zip file to be decompressed
+        """
+        with zipfile.ZipFile(filename, "r") as zip_file:
+            zip_file.extractall()
+
+    def load_model(
+        self, 
+        model_file: str = "model_10_202405.ubj",
+        file_id: str = "1O34gyQ67rvrP9VFkNaagTDM9IP4iqAjM"
+    ) -> XGBRegressor:
+        """
+        Download and decompress model from Google Drive
+        Parameters
+        ----------
+        model_file: str
+            The name of the model as string
+        file_id: str
+            Google id to download the file
+        """
+        zipfile_model = model_file + ".zip"
+
+        if not os.path.isfile(zipfile_model):
+            print("Downloading model ...")
+            self._download_model(zipfile_model, file_id)
+        if not os.path.isfile(model_file):
+            print("Preparing model ...")
+            self._decompress_zipfile(zipfile_model)
+        print("Loading model ...")
+        loaded_model = XGBRegressor()
+        loaded_model.load_model(model_file)
+        self.model = loaded_model 
+
+    def get_data(
+        self,
+        latitude: float,
+        longitude: float,
+        start_date: str,
+        kwp: float,
+        orientation: float,
+        tilt: float,
```

Comment:
is it worth adding some default values fomr orientation and tilt?

Reply:
tilt=30? orientation =180?

Reply:
Added